source/slang/type-layout.cpp

```diff
@@ -1599,6 +1604,28 @@ SimpleLayoutInfo GetLayoutImpl(
 
             return info;
         }
+        else if (auto globalGenParam = declRef.As<GlobalGenericParamDecl>())
+        {
+            SimpleLayoutInfo info;
+            info.alignment = 0;
+            info.size = 0;
+            info.kind = LayoutResourceKind::GenericResource;
```

Comment:
Oh wait! I'm completely misunderstanding. You are using this resource to represent "this type depends on a generic type parameter, and so we can't do layout on it right now".

Aside: I suppose we should technically be catching `GenericTypeParamDecl` in this case as well, even though that can't come up just yet.